tokio-util/src/io/stream_reader.rs

```diff
@@ -94,10 +92,10 @@ where
     }
 }
 
-impl<S, B> AsyncRead for StreamReader<S, B>
+impl<S: TryStream> AsyncRead for StreamReader<S>
```

Comment:
are we using `TryStream` in other places or did we decide to go against it?

Reply:
I don't think we use it anywhere else, but we also use other weird traits in `tokio-util` such as `Sink`. The reason I added it here is that it removes the need for two generic parameters.

Reply:
Hmmm yeah, I know it removes the need but there might be some coherence issues. I am happy to go with it here since this is the util crate.

Reply:
I changed my mind on this issue.